include/sampling/mmcs.hpp

```diff
@@ -0,0 +1,129 @@
+// VolEsti (volume computation and sampling library)
+// Copyright (c) 2021 Vissarion Fisikopoulos
+// Copyright (c) 2021 Apostolos Chalkis
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef MMCS_HPP
+#define MMCS_HPP
+
+#include "diagnostics/ess_window_updater.hpp"
+
+/**
+   The class implements a single step of the Multiphase Monte Carlo Sampling algorithm
+   given in,
+
+   A. Chalkis, V. Fisikopoulos, E. Tsigaridas, H. Zafeiropoulos, Geometric algorithms for sampling the flux space of metabolic networks, SoCG 21.
+
+ * @tparam Polytope convex polytope type
+ * @tparam RandomNumberGenerator random number generator type
+ * @tparam MT matrix type
+ * @tparam Point cartensian point type
+ * @tparam WalkTypePolicy random walk type
+*/
+template 
+<
+        typename Polytope,
+        typename RandomNumberGenerator,
+        typename MT,
+        typename Point,
+        typename WalkTypePolicy
+>
+void perform_mmcs_step(Polytope &P,
+                       RandomNumberGenerator &rng,
+                       const unsigned int &walk_length,
```

Comment:
should this be `unsigned int const&` as below ?

Reply:
done